### DISCUSSION TITLE
Using nuxt module in application without calling prepack each time

### DISCUSSION THREAD

**Discussion by mklueh** [nuxt/nuxt#16751] [Questions] [answered]
Hello,

I'm having an NX workspace with one Nuxt 3 application and one custom Nuxt module library.

It seems whenever you change something in the nuxt module, you have to call prepack before running your actual application again.

Is there a way to just use the module from it's source code without packing it?

**Comment by cpreston321** [comment, marked as answer]
@mklueh if you use `nuxt-module-build --stub` it will generate the paths so you don't have to build it every time in development.

Example of the output in the `dist/module.mjs` file.

```js
import jiti from 'jiti';
export default jiti(null, { interopDefault: true })('/Users/..../..../projects/nuxt-swiper/src/module');
```

but when you run `nuxt-module-build` the output will driven for production and will have to be re-compiled every time. This is what it should look like when running the command with output: `dist/module.mjs` 

<img width="999" alt="image" src="https://user-images.githubusercontent.com/6619884/192374790-0b009044-b11a-4fb1-959b-c04e8de7f8cf.png">

Hopefully this helps you! 

Thanks 💚

**Reply by cpreston321** [reply]
`jitti` is runtime so every time you make changes it should recompile it on whatever is using that module during development.

**Reply by cpreston321** [reply]
I also don't know if you are using `nuxt-module-build` in your workspace but you could reference the repo here to see the commands and packages installed for your module development [here](https://github.com/nuxt/module-builder)

**Reply by mklueh** [reply]
Thank you, that indeed works!! 
I'm using nuxt-module-build, but not sure what you mean by "in your workspace". It's an NX workspace and the @nuxt/module-builder dependency is installed in the root of the project, if you mean that.

How would I deal with it during CI? Just run nuxt-module-build before building my actual application?

**Reply by cpreston321** [reply]
Correct! That is how I would deal with it. 

Also if this helped you I would mark this `answer` as complete to help others in the future 🤘

**Reply by mklueh** [reply]
I found out that with NX you can tell it to automatically build dependencies of the application


```
"nx": {
    "targets": {
      "build": {
        "outputs": [
          "apps/my-app/dist",
          "apps/my-app/.output"
        ],
        "dependsOn": [
          "^prepack"
        ]
      }
    }
  }
```

This is amazing, because no matter how many modules you add, it will ensure all are prepacked without changing any configuration anymore.

Thanks for your help!
